### PR TITLE
Add high-level overview of Dart

### PR DESCRIPTION
**Pull request by timsneath**
The left sidebar of the Flutter docs page has a prominent link to a Dart overview:

image

but the overview there is a [rather uneven list of resources](https://dart.dev/guides/language), from great stuff like the [language tour](https://dart.dev/guides/language/language-tour) to ancient content about [migrating to sound types](https://dart.dev/guides/language/sound-problems) and [language specifications](https://dart.dev/guides/language/spec). 

This article attempts to rectify this with a higher-altitude, more general page that tours Dart through three lenses: the language, the libraries and the compilers/runtime.

### CODE REVIEW

**Comment by kwalrath** [comment]
Looking now... It seems that Dart-logo_lockup_dart_horizontal.png is missing?

**Comment by timsneath** [comment]
It's here: https://github.com/dart-lang/site-www/blob/e5434789b8b2776a03e8e38465b0684d85e55676/src/_assets/image/logo_lockup_dart_horizontal.png but I might have got the wrong URL?

**Comment by kwalrath** [comment]
> It's here: https://github.com/dart-lang/site-www/blob/e5434789b8b2776a03e8e38465b0684d85e55676/src/_assets/image/logo_lockup_dart_horizontal.png but I might have got the wrong URL?

If your fix didn't work, I can fix it when I do the copy edit (which I'm starting now...).

**Comment by kwalrath** [comment]
Staged here: https://kw-staging-dartlang-2.firebaseapp.com/overview

Before I start with a copy edit, I have a couple of questions:

* [Dart: The Platform](https://kw-staging-dartlang-2.firebaseapp.com/overview#dart-the-platform) is so visually similar to https://kw-staging-dartlang-2.firebaseapp.com/platforms that it's a little disorienting. What would be a good way to differentiate them?
* The [library section](https://kw-staging-dartlang-2.firebaseapp.com/overview#dart-the-libraries) could be easier to read. Should we use a table? Boldface? Maybe it should have less detail? Also, should this section link to https://dart.dev/guides/libraries and https://dart.dev/guides/libraries/useful-libraries, and https://dart.dev/guides/libraries/library-tour? (If not... what does that say about those pages?)

**Comment by timsneath** [comment]
Great -- thanks so much. 

This is intended as the one-pager précis of all these topics, so some of this text is drawn from the other pages. I deliberately put the diagram below the fold, but we could remove it altogether, obviously. 

Those other library pages are very linkable, yes. Now I can see the page on a staging server, I'll make some tweaks.

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:11]: It looks a bit odd with the logo hanging here in the middle of the page. Perhaps we can pull it up into the heading of the page?

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:8]: I would link 'platform' here to https://dart.dev/platforms

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:14]: Consider s/development/their development/

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:6]: I wonder if it's time to reconsider out discussion about "client" as our label. As mentioned before I'd feel more comfortable with something like 'app-optimized'.

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:27]: I think we should pull the "runtime checks" part of of this sentence, and then we can state in a later sentence: The Dart typing system is also flexible, allowing the use of a 'dynamic' type combined with runtime checks, which can be useful during experimentation or for code that needs to be especially dynamic.

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:31]: Link 'sound null safety' to https://dart.dev/null-safety

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:33]: Consider s/code analysis/static code analysis/

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:38]: It feels like some kind of intro sentence is needed before we dive into this code snippet

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:39]: This should be an embedded DartPad

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:68]: We should update this to null safety; in that, this needs to be `[int? seed]`. We could add a comment that explains that all other variable in this code are non-null.

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:86]: Link 'core libraries' to https://dart.dev/guides/libraries

File: src/overview.md
Comment (timsneath) [reply src/overview.md:11]: Agree with you. I've floated it to the right now using my amazing CSS talents :)

File: src/overview.md
Comment (timsneath) [reply src/overview.md:6]: Yeah, but "app-optimized" isn't specific enough. Server apps, cloud apps, client apps, etc. I'm resolving this conversation, but we can continue it offline to see if we can find something workable.

**Comment by timsneath** [comment]
@mit-mit @kwalrath Thanks so much for the great review comments. I've updated the article per the various comments -- adding links where appropriate, fixing the image, adding additional explanatory text. Would love you to take a second look if that's OK?

**Comment by mit-mit** [comment]
Thanks Tim, I'll take a pass at this after we have beta out

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:12]: I think we should remove the "across Google" part.

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:38]: Consider emphasizing 'always' with italic

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:43]: Comma before and (twice)

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:60]: Maybe add a comment that refers back to the type discussion above:

`var total - 0; // The type of var is inferred to be int.`

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:92]: This can be https://dartpad.dev/b2e3a914bd39d0408d0580c42ef2e58b?null_safety=true

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:155]: Comma before and

**Review by mit-mit — APPROVED** [review]
LGTM with a few nit comments

File: src/overview.md
Comment (timsneath) [reply src/overview.md]: Former is deliberate (it's one clause), fixed the latter.

**Comment by timsneath** [comment]
Thanks! @kwalrath are we good to go from your side?

**Comment by kwalrath** [comment]
> Thanks! @kwalrath are we good to go from your side?

Not quite yet. I'll do a wordsmithing pass now (was waiting for the technical reviews first).

**Comment by kwalrath** [comment]
We'll need to have this in the navigation somewhere (side or top), and perhaps link to it from the homepage as well. How do you think we should point to this?

**Comment by mit-mit** [comment]
Add an Overview entry at the top under Language, like we have under Core libraries?

**Comment by kwalrath** [comment]
> Add an Overview entry at the top under Language, like we have under Core libraries?

That would be easiest. My one hesitation is that the scope of the paper seems larger. It starts with "Dart is a client-optimized language", but it has separate sections for the language, libraries, and platform. In fact, its platforms section is basically the top of https://dart.dev/platforms.

Maybe we should get rid of the Platforms page and substitute this page for it, with a redirect from /platforms to /overview#dart-the-platform. In that case, the top nav might be something like this:

```
Overview    Docs    Community   Try Dart    Get Dart
```

File: src/overview.md
Comment (kwalrath) [reply src/overview.md:49]: > This should be an embedded DartPad

That was my thought, too. Is there a good reason *not* to make this an embedded DartPad?

**Review by kwalrath — CHANGES_REQUESTED** [review]
We need to decide on navigation and whether this replaces the Platforms page. Then I need to do a proofreading pass.

**Comment by timsneath** [comment]
@kwalrath just checking in to make sure the next step is clear -- is the navigation question for @mit-mit or is it a question that you're pondering yourself?

**Comment by kwalrath** [comment]
> @kwalrath just checking in to make sure the next step is clear -- is the navigation question for @mit-mit or is it a question that you're pondering yourself?

I think it's a question for both @mit-mit and you, @timsneath. Once we know the intended scope of the page, I can edit it.

File: src/overview.md
Comment (mit-mit) [inline src/overview.md:135]: The `compression` link is broken; I think it's the linefeed that needs to be removed

**Comment by mit-mit** [comment]
If we bring over a bit of the text from https://kw-staging-dartlang-2.firebaseapp.com/platforms into the Overview: platforms section, then I like the solution you mention above Kathy with replacing the Platforms link in the top-nav with Overview.

**Review by mit-mit — CHANGES_REQUESTED** [review]

**Comment by mit-mit** [comment]
@kwalrath how about we go ahead and merge this, and then I'll take a stab at making this replace the existing Platforms page in the top-nav?

**Comment by kwalrath** [comment]
> @kwalrath how about we go ahead and merge this, and then I'll take a stab at making this replace the existing Platforms page in the top-nav?

That sounds reasonable. I'll get started on the copy edit.

File: src/overview.md
Comment (kwalrath) [inline src/overview.md:151]: Should this be "that's supported on [all major browsers](/faq#q-what-browsers-do-you-support-as-javascript-compilation-targets)?

Because we technically don't support all major browsers, according to https://dart.dev/faq#q-what-browsers-do-you-support-as-javascript-compilation-targets.

File: src/overview.md
Comment (kwalrath) [inline src/overview.md:58]: Should we be linking to dart.dev/samples instead? We know that'll stay updated.

/cc @johnpryan

**Comment by kwalrath** [comment]
I've updated https://kw-staging-dartlang-2.firebaseapp.com/overview. Here are the remaining issues that I'm aware of (none major):

* Is there a good reason not to embed the sample code in a DartPad?
* What code examples should we link to after the sample code?
* What should we say about browser support for Dart compiled to JavaScript?
* PUNTED: Link to this page in the navigation, somehow.
* One last copy edit pass, once all the text is ready. Stuff like:
  * Macros (e.g. `{{site.pub-pkg}}` instead of `https://pub.dev/packages`)
  * Terminology consistency (adding terms to go/dart-terms as necessary)
  * Polish (e.g. removing some more passive voice, readability tweaks, maybe doing an auto-translate from and back to English)

**Comment by mit-mit** [comment]
> Is there a good reason not to embed the sample code in a DartPad?

No, would be good to use an embedded DartPat.

> What code examples should we link to after the sample code?

Maybe just link to https://kw-staging-dartlang-2.firebaseapp.com/samples ?

> What should we say about browser support for Dart compiled to JavaScript?

I don't think we should say anything here in an Overview. That should be in the systems requirements page we've discussed elsewhere.

> PUNTED: Link to this page in the navigation, somehow.

I'll do that in a follow-on PR adding this as Overview in the top-nav.

> One last copy edit pass, once all the text is ready. Stuff like:

SGTM, thanks!

File: src/overview.md
Comment (johnpryan) [reply src/overview.md:58]: +1 - I haven't updated Dart By Example. (it's somewhere on my list, but not very high up right now)

Comment (timsneath) [reply src/overview.md:58]: I just love it though, John -- it's such a comprehensive list of examples (compared to the language tour). FWIW, I'd love to see an extended version of this on the website.

Comment (mit-mit) [reply src/overview.md:58]: John is going to take a stab at combing his content with the current Dart samples content, and maybe even in the process wire the samples up with embedded DartPad 🚀

**Comment by kwalrath** [comment]
OK, I'm done. @timsneath are my changes OK by you? If so, let's ship this!

https://kw-staging-dartlang-2.firebaseapp.com/overview

**Review by kwalrath — APPROVED** [review]

**Comment by timsneath** [comment]
LGTM! When it's live we can update the Flutter side to point to it.

**Comment by kwalrath** [comment]
If Travis doesn't auto-push this, I'll try to do that by hand later today.

(Because I have no faith in Travis, I ran the pre-push script locally; it passed.)

**Comment by kwalrath** [comment]
Travis tanked, so I pushed by hand. It's up! https://dart.dev/overview

**Pull request merged**
